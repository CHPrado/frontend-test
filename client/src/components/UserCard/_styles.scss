li {
  background: #FFF;
  border: 2px solid #FFF;
  border-radius: 8px;
  padding: 32px 24px 16px;

  @extend .flex;
  flex-direction: column;
  text-align: center;

  img {
    border-radius: 10rem;
    width: 300px;
  }

  span {
    margin-top: 12px;
    color: $text-color;
    @extend .flex;
  }

  button {
    margin-top: 12px;
    width: 100%;
    height: 50px;
    background: transparent;
    border-radius: 8px;
    transition: background-color 0.2s;
    border: solid $text-color;

    cursor: pointer;

    svg {
      color: $text-color;
    }
  }

  button:hover {
    border: solid #e00e0e;

    svg {
      color: #e00e0e;
    }
  }
}