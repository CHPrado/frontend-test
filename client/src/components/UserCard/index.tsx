import React from 'react';
import { FiTrash2 } from 'react-icons/fi';

import { UserProps } from '../../interfaces';

interface UserCardProps {
  user: UserProps;
  handleDeleteClick: () => void;
}

const UserCard: React.FC<UserCardProps> = ({ user, handleDeleteClick }) => (
  <li>
    <img src={user.avatar} alt={user.username} />
    
    <span>{user.name}</span>
    <span>{user.username}</span>
    <span>{user.email}</span>
    <span>{user.phone}</span>

    <button onClick={handleDeleteClick} type="button">
      <FiTrash2 size={20} />
    </button>
  </li>
);

export default UserCard;