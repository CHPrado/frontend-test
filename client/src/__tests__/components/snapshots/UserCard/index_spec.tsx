import React from 'react';
import renderer from 'react-test-renderer';

import UserCard from '../../../../components/UserCard';

test('it matches the snapshot', () => {
  const user = {
    id: 1,
    name: 'Test Name',
    username: 'Test Username',
    email: 'teste@email.com',
    phone: '123456',
    avatar: 'teste-avatar.png',
  }
  const tree = renderer.create(
    <UserCard user={user} handleDeleteClick={() => {}} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
