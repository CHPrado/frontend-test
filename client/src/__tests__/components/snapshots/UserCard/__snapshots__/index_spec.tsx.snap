// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<li>
  <img
    alt="Test Username"
    src="teste-avatar.png"
  />
  <span>
    Test Name
  </span>
  <span>
    Test Username
  </span>
  <span>
    teste@email.com
  </span>
  <span>
    123456
  </span>
  <button
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="none"
      height={20}
      size={20}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="3 6 5 6 21 6"
      />
      <path
        d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
      />
      <line
        x1="10"
        x2="10"
        y1="11"
        y2="17"
      />
      <line
        x1="14"
        x2="14"
        y1="11"
        y2="17"
      />
    </svg>
  </button>
</li>
`;
